packages/terra-list/src/SelectableUtils.js

```diff
@@ -86,6 +86,76 @@ const shouldHandleMultiSelect = (children, maxSelectionCount, currentIndexes, ne
  */
 const shouldHandleSingleSelect = (currentIndex, newIndex) => newIndex !== currentIndex;
 
+const wrappedOnClickForItem = (item, index, handleOnChange) => {
+  const initialOnClick = item.props.onClick;
+
+  return (event) => {
+    // The default isSelectable attribute is either undefined or true, unless the consumer specifies the item isSelectable attribute as false.
+    if (item.props.isSelectable !== false && handleOnChange) {
+      handleOnChange(event, index);
```

Comment:
IMO it'd be less confusing to just call `handleOnChange` `onChange` given this method is now outside of the selectable components and we directly pass the onChange provided to the selectableList.